src/pages/CreateAnalysisPage/RightArrowButton.tsx

```diff
@@ -3,16 +3,28 @@ import { ArrowRightIcon } from '@patternfly/react-icons';
 import React from "react";
 
 interface RightArrowButtonProps { 
-  click: () => void 
+  click: (XrayModel: string | void, CTModel: string | void) => (() => void) // Factory function to take in the chosen models and pass a function that will have the correct state in the history hook
   className?: string
+  XrayModel?: string
+  CTModel?: string
 }
 
-const RightArrowButton: React.FC<RightArrowButtonProps> = ({click, className, children}) => {
-  return (
-    <Button className={className} variant="primary" onClick={click}>
-      {children} <ArrowRightIcon></ArrowRightIcon>
-    </Button>
-  )
+const RightArrowButton: React.FC<RightArrowButtonProps> = ({click, XrayModel, CTModel, className, children, }) => {
+
+  if (XrayModel || CTModel) {
+    return (
+      <Button className={className} variant="primary" onClick={click(XrayModel, CTModel)}>
+        {children} <ArrowRightIcon></ArrowRightIcon>
+      </Button>
+    )
+  } else {
+    return (
+      <Button className={className} variant="primary" onClick={click()}>
+        {children} <ArrowRightIcon></ArrowRightIcon>
+      </Button>
+    )
```

Comment:
From my understanding, `RightArrowButton` is a shared component that is used in multiple other components. By adding `XrayModel` and `CTModel`, in my opinion it makes this component more into Analyze button component since it has specifics of the analyze button implementation.

I would suggest keeping this component as the general right arrow component and create a wrapper function that passes in the XrayModel and CTModel when submitAnalysis is called. Please, look into my comment in CreateAnalysisDetail.tsx for more information.